File: solutions/0796. Rotate String/0796_1.cpp
// Time complexity: O(n^2)
// Space complexity: O(1)
class Solution {
public:
    bool rotateString(const string &s, const string &goal) {
        if (s.size() != goal.size()) {
            return false;
        }
        int n = s.size();
        for (int i = 0; i < n; ++i) {
            bool good = true;
            for (int j = 0; j < n; ++j) {
                int index = (i + j) % n;
                if (s[index] != goal[j]) {
                    good = false; 
                    break
                }
            }
            if (good) {
                return true;
            }
        }
        return false;
    }
};

File: solutions/0796. Rotate String/0796_1.py
# Time complexity: O(n^2)
# Space complexity: O(1)
class Solution:
    def rotateString(self, s: str, goal: str) -> bool:
        if len(s) != len(goal):
            return False

        for shift in range(len(s)):
            if all(s[(shift + i) % len(s)] == goal[i] for i in range(len(s))):
                return True
        return False    